lib/api/redfish-1.0/chassis.js

```diff
@@ -286,36 +311,82 @@ var getThermal = controller(function(req, res) {
     var systemId = req.swagger.params.identifier.value.split('.')[1];
     var options = redfish.makeOptions(req, res, req.swagger.params.identifier.value);
 
-    return dataFactory(chassisId, 'system')
-    .then(function(system) {
-        return Promise.all([
-            dataFactory(systemId || system, 'sdrData'),
-            waterline.nodes.findByIdentifier(systemId || system)
-            .then(function(obj) {
-                return env.get('config.redfish', '{}', obj && obj.sku ? [ obj.sku ] : null );
+    return redfish.getVendorNameById(chassisId)
+    .then(function(result) {
+        if (result.vendor === 'Cisco') {
+            return dataFactory(chassisId, 'ucsThermal')
+            .then(function(thermalData) {
+                var ucsPowerThermalData = thermalData.ucsPowerThermalData;
+                var ucsFanData = thermalData.ucsFanData;
+                var regxObj = /.*_temp_sen_.*[^avg|max|min]$/;
+                options.tempList = [];
+                options.fanList = [];
+                _.forEach(ucsPowerThermalData, function(data) {
+                    _.forEach(data, function(item) {
+                        _.forEach(_.keys(item), function(key) {
+                            if(regxObj.test(key)) {
+                                options.tempList.push(
+                                    {
+                                        'sensorId': key.match(regxObj)[0],
```

Comment:
From what I learnt, this key.match(regxObj)[0] is not necessary as it returns key.